providers/implementations/keymgmt/ml_kem_kmgmt.c

```diff
@@ -199,29 +204,36 @@ static int ml_kem_export(void *vkey, int selection, OSSL_CALLBACK *param_cb,
          * ossl_param_build_set_octet_string(), which will then also use the
          * secure heap.
          */
-        prvenc = OPENSSL_secure_zalloc(v->prvkey_bytes);
-        if (prvenc == NULL)
-            goto err;
+        if (ossl_ml_kem_have_seed(key)) {
+            prvlen = ML_KEM_SEED_BYTES;
+            if ((prvenc = OPENSSL_secure_zalloc(prvlen)) == NULL
+                || !ossl_ml_kem_encode_key_seed(prvenc, prvlen, key))
+                goto err;
+        } else {
+            prvlen = v->prvkey_bytes;
+            if ((prvenc = OPENSSL_secure_zalloc(prvlen)) == NULL
+                || !ossl_ml_kem_encode_private_key(prvenc, prvlen, key))
+                goto err;
```

Comment:
Did we not conclude that the private key encoding on export would always be `dk` and we would have a separate `seed` param? I assume this will be resolved in some subsequent PR

Reply:
My understanding was that for now the output defaults to the seed (when available), and **iff** there's a path forward where we can output both, then we'd output both (when both are available).

Reply:
I think you are confusing the encoder output format with the *export*. We are talking specifically about the export here, and I thought we had said that `OSSL_PKEY_PARAM_PRIV_KEY` would *always* be `dk` and there would be some other param for the seed.

Reply:
No, you may have gotten that impression, but that's not the case at this time...